Fix issues with failed JSON parsing in JS API responses

WizardCM opened the pull request:
### Description

Using a [forum post](https://magpcss.org/ceforum/viewtopic.php?f=10&t=14807#p33456) as reference, overhauled JSON handling internally to use CefValue rather than the V8 context.

Note: It seems the V8 context is still required in this code block. I am unsure exactly why (it crashes otherwise).

### Motivation and Context

When users call API functions, they expect a response. Especially a valid one. Fixes #390.

### How Has This Been Tested?

`obsstudio.getScenes` and `obsstudio.getCurrentScene` with scene names that contain characters such as `"` and `\`.

Also re-tested all other get functions to make sure nothing else was broken.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

tytan652 commented on browser-client.cpp:
```diff
@@ -206,13 +206,13 @@ bool BrowserClient::OnProcessMessageReceived(
 					obs_source_get_name(source));
 			}
-			json = scenes_vector;
+			json = Json::array(scenes_vector);
```
nlohmann is smart enough to convert a vector of JSONs to a JSON array of objects. And `Json::array` is json11 only.
```suggestion
			json = scenes_vector;
```

WizardCM replied:
Confirmed. Scrapped the change on my side.

tytan652 commented on browser-client.cpp:
```diff
@@ -234,13 +234,13 @@ bool BrowserClient::OnProcessMessageReceived(
 					obs_source_get_name(source));
 			}
-			json = transitions_vector;
+			json = Json::array(transitions_vector);
```
nlohmann is smart enough to convert a vector of JSONs to a JSON array of objects. And `Json::array` is json11 only.
```suggestion
			json = transitions_vector;
```

WizardCM replied:
Confirmed. Scrapped the change on my side.

tytan652 reviewed (commented):
You might need to check if conversion are done correctly. But it should be good once the added json11 stuff is removed.

WizardCM commented:
Rebased on master (after fixing Nlohmann JSON compilation on Windows & macOS), removed unnecessary changes based on @tytan652's feedback, and retested with Nlohmann JSON specifically. It has the same output difference as json11 so this PR is still necessary to fix the linked issue.

Ready to be reviewed.
